### PR TITLE
Project library Axel & Lovisa

### PR DESCRIPTION
**Pull request by AHPIXI**
## Netlify link
Add your Netlify link here. https://sensational-parfait-a86405.netlify.app/
PS. Don't forget to add it in your readme as well.

## Collaborators
Add your collaborators here. Write their GitHub usernames in square brackets. If there's more than one, separate them with a comma, like this:
[Lovisaaberg]

### CODE REVIEW

File: .DS_Store
Comment (NellaMori) [inline .DS_Store:1]: Please add .DS_Store to your .gitignore file (if you don't have one, you need to create it) 👍

File: code/.DS_Store
Comment (NellaMori) [inline code/.DS_Store:1]: same here 🧹

File: code/index.html [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    <header>
+      <h1>Project Library</h1>
+    </header>
+    <div class="wrapper">
+      <div id="filter-books" class="filter-box">
+        <p class="headings-top">Filter books by genre</p>
+        <button id="all" class="filter-buttons">All</button>
+        <button id="fantasy" class="filter-buttons">Fantasy</button>
+        <button id="horror" class="filter-buttons">Horror</button>
+        <button id="adventure" class="filter-buttons">Adventure</button>
```
Comment (NellaMori) [inline code/index.html:19]: no need to pass an `id` if you're not using it on your stylesheet, keep only valuable code 🧹

File: code/style.css [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  padding: 8px 16px 8px 16px;
+  border-radius: 50px;
+  border: none;
+}
+
+.sort-buttons:hover {
+  border: 2px solid #0018a4;
+  color: white;
+  background-color: #ff6589;
+}
```
Comment (NellaMori) [inline code/style.css]: when applying a board on hovering, the surrounding UI is changing to make space for it and this is to be avoided. A solution would be to wrap the button in a div and give the div a fixed height 👍

File: code/style.css [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+  padding: 8px 16px 8px 16px;
+  font-style: bold;
+  border-radius: 50px;
+  border: none;
+}
+
+.filter-buttons:hover,
+#random:hover {
+  border: 2px solid #0018a4;
+}
```
Comment (NellaMori) [inline code/style.css]: same here. Hover effects should not affect the positioning of the rest of the UI

**Review by NellaMori — APPROVED** [review]
Hi Axel & Lovisa, congrats you did a really good job with this project 🔥 You followed the instructions meticulously, implemented basic requirements and even followed the design to inspire the UI of your app. Overall a very good execution 👍 I left some comments around the code for little improvements related to the UI, keep up the great work! 🏅

File: code/script.js [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+const displayBooks = (bookBoxes) => {
+  container.innerHTML = "";
+
+  bookBoxes.forEach((books) => {
+    container.innerHTML += `
+    <div class="book-cards">
+      <img class="img-cards" src="${books.image}">
+      <h2>${books.title}</h2>
+      <p class="subheadings">${books.author}</p>
+      <p class="subheadings"${books.year}</p>
```
Comment (Tejpex) [inline code/script.js]: You're missing an > which makes the year not showing on the webpage.

File: code/script.js [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+const filterFantasy = document.getElementById("fantasy");
+const filterHorror = document.getElementById("horror");
+const filterAdventure = document.getElementById("adventure");
+const sortBooksAZ = document.getElementById("az");
+const sortBooksZA = document.getElementById("za");
+const sortBooksNewest = document.getElementById("newest");
+const sortBooksOldest = document.getElementById("oldest");
+const sortBooksRating = document.getElementById("highestRating");
+const randomButton = document.getElementById("random");
+const searchBar = document.getElementById("search-bar");
```
Comment (Tejpex) [inline code/script.js]: Is this DOM-selector being used?

File: code/script.js [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+
+const adventureBooks = (filterBooks) => {
+  const filterAdventure = books.filter((bookGenres) => {
+    return bookGenres.genre === "Adventure";
+  });
+  displayBooks(filterAdventure);
+};
+
+// Function and eventlistener for a random button generator
+randomButton.addEventListener("click", () => {
```
Comment (Tejpex) [inline code/script.js:272]: Great that the random function works!

File: code/script.js [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+      <p class="subheadings">${books.genre}</p>
+      <p class="subheadings">⭐️ ${books.rating}</p>
+      <p class="description">${books.description}</p>
+    </div>
+    `;
+  });
+};
+
+//Functions to sort the books alphabetically, by publishing year & highest rating
+const sortBooksAtoZ = () => {
```
Comment (Tejpex) [inline code/script.js:217]: Good that the function is short and easy to read.

File: code/script.js [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+  });
+  displayBooks(filterHorror);
+};
+
+const adventureBooks = (filterBooks) => {
+  const filterAdventure = books.filter((bookGenres) => {
+    return bookGenres.genre === "Adventure";
+  });
+  displayBooks(filterAdventure);
+};
```
Comment (Tejpex) [inline code/script.js:269]: Maybe this is a shorter way to do the same thing:

```javascript
const filterBooks = () => {
  const value = filterDropdown.value;

  if (value === "all") {
    displayBooks(books);
  } else {
    const filteredList = books.filter((book) => {
      return book.genre === value;
    });
    displayBooks(filteredList);
  }
};
```

**Review by Tejpex — COMMENTED** [review]
Overall easy to read code. Good looking page with working functions. Only minor fixes found.

/Erika, Katarina, Alexandra

File: code/index.html (hunk 1 above)
Comment (Lovisaaberg) [reply code/index.html:19]: We used the id for the DOM selectors in the js file, and the class for styling in CSS.

File: code/style.css (hunk 2 above)
Comment (Lovisaaberg) [reply code/style.css]: Changed it to outline instead 👍

**Pull request closed without merging**